projects/cashmere/src/lib/pipes/file-size/file-size.pipe.ts

```diff
@@ -0,0 +1,51 @@
+import {Pipe, PipeTransform} from '@angular/core';
+
+// Convert bytes into largest possible unit.
+// Credit to https://gist.github.com/JonCatmull/ecdf9441aaa37336d9ae2c7f9cb7289a
+@Pipe({name: 'fileSize', pure: true})
+export class FileSizePipe implements PipeTransform {
+    private units: string[] = ['bytes', 'KB', 'MB', 'GB', 'TB', 'PB'];
```

Comment:
@isaaclyman Currently the largest unit is petabyte, which should exceed our needs for now and the near future.  If it exceeds 1024 petabytes, it displays as number of petabytes (i.e. `239847573 PB`)